src/main/scala/org/stormenroute/mecha/package.scala

```diff
@@ -432,4 +432,66 @@ package object mecha {
     def default(v: =>T): Input.Query[T] = Input.default(query)(v)
   }
 
+  private[mecha] object GitIgnore {
+    import java.nio.charset.StandardCharsets._
+    import java.nio.file.{FileSystems, Paths, Files, Path}
+    import java.nio.file.StandardOpenOption._
+    import scala.collection.JavaConversions._
+
+    val log = MechaLog.Println
+
+    def ignore(ignoreFilePath: Path, toAdd: String) = {
+      log.info(s"Adding '$toAdd' to '${ignoreFilePath.toString}'...")
+      writeFile(ignoreFilePath, add(readFile(ignoreFilePath), toAdd))
+    }
+
+    private def readFile(file: Path): Seq[String] = {
+      Files.readAllLines(file, UTF_8)
+    }
+
+    private def writeFile(file: Path, content: Seq[String]) = {
+      Files.write(file, content, UTF_8, WRITE, CREATE)
+    }
+
+    private def add(existing: Seq[String], `new`: String): Seq[String] = {
+      val patterns = existing.map(line).collect({case p: Pattern => p})
+      val alreadyIgnored = patterns.find(p => p.matches(`new`))
+      if (alreadyIgnored.isDefined) {
+        log.info(s"'${`new`}' is already ignored via pattern '${alreadyIgnored.get.p}'")
+        existing
+      }
+      else {
+        existing :+ `new`
+      }
+    }
+
+    private def line(l: String): Line = l.trim match {
+      case "" => Empty
+      case s if s.startsWith("#") => Comment(s)
+      case _ => Pattern(l)
+    }
+
+    sealed trait Line {
+      def matches(path: String): Boolean = false
+    }
+    object Empty extends Line
+    case class Comment(c: String) extends Line
+    case class Pattern(p: String) extends Line {
+      val trimmed = p.trim
+      val negative = trimmed.startsWith("!")
+      lazy val matcher = FileSystems.getDefault.getPathMatcher(s"glob:./${cleanUp(trimmed)}")
+      override def matches(dirName: String): Boolean = {
+        val path = if (!dirName.startsWith("./")) s"./$dirName" else dirName
+        val matches = matcher.matches(Paths.get(path))
+        (!negative && matches) || (negative && !matches)
```

Comment:
Cool - thanks!

Btw, thinking a bit about the case where the folder that you're after is whitelisted, it seems to me that the best option is to report an error and disallow tracking until the problematic rules is fixed manually by the dev.